### PR TITLE
Remove article model logic

### PR DESCRIPTION
**Pull request by andygnewman**
Objectives:
- remove article card data manipulation logic from n-section
- migrate this logic to n-content-model (see this [PR](https://github.com/Financial-Times/n-content-model/pull/11))
- retain only code that is directly related to display characteristics

(Ideally even the stuff that is retained here should be in n-card rather than n-section, but taking this one step at a time)

### CODE REVIEW

File: src/models/article.js [hunk 1]
```diff
@@ -1,110 +1,69 @@
...
+		{
+			type: item.type,
+			id: item.id,
+			title: item.title,
+			published: item.published,
+			lastPublished: item.lastPublished
+		}
+	);
+
+	return content;
```
Comment (ironsidevsquincy) [inline src/models/article.js]: what's the rationale for not moving everything in here into n-content-model?

Comment (andygnewman) [inline src/models/article.js]: I've left what is essentially to do with display here, ie. driven by the options passed into n-section, now documented in the wiki - https://github.com/Financial-Times/n-section/wiki/Content-Display-Options

File: src/models/article.js [hunk 2]
```diff
@@ -1,110 +1,69 @@
...
+			content.brand = item.branding;
 		} else {
-			article.brand = {
+			content.brand = {
 				title: 'Opinion',
 				url: '/stream/sectionsId/MTE2-U2VjdGlvbnM=',
-				type: 'opinion'
 			};
 		}
+		content.brand.type = 'opinion';
```
Comment (ironsidevsquincy) [inline src/models/article.js]: this is data manipulation - should probably live in n-content-model?

File: src/models/article.js [hunk 3]
```diff
@@ -1,110 +1,69 @@
...
-		article.isMain = true;
-	}
-	if (type === 'editors-pick') {
-		article.brand = {
+	if (item.isEditorsChoice) {
+		content.brand = {
 			title: 'Editor’s Pick',
 			type: 'editors-pick'
 		};
 	}
```
Comment (ironsidevsquincy) [inline src/models/article.js]: ditto?

File: src/models/article.js [hunk 4]
```diff
@@ -1,110 +1,69 @@
...
+		content.related = Object.assign({},
+			{
+				items: item.relatedContent
+					.filter(relatedItem => relatedItem.id !== item.id)
+					.slice(0, 3)
+			}, opts.related);
+	}
+
+	// hide timestamp
+	if (item.type === 'fast-ft') content.hideTimestampState = true;
```
Comment (ironsidevsquincy) [inline src/models/article.js]: ditto?

File: src/models/article.js (hunk 1 above)
Comment (ironsidevsquincy) [inline src/models/article.js]: Few things seem like they should live in the model code

File: src/models/article.js [hunk 5]
```diff
@@ -1,111 +1,49 @@
...
+			title: item.title,
+			published: item.published,
+			lastPublished: item.lastPublished,
+			hideTimestampState: item.hideTimestampState,
+			liveBlog: item.liveBlog,
+			subheading: item.subheading,
+			type: item.type,
+			brand: item.brand,
+			branding: item.branding
+		}
```
Comment (ironsidevsquincy) [inline src/models/article.js]: these can just be a single object

**Comment by ironsidevsquincy** [comment]
👍

File: src/models/article.js (hunk 5 above)
Comment (andygnewman) [inline src/models/article.js]: Good spot - had them as separate that reflected their different origins but quite correct that they can be a single object.

**Pull request merged**